### ISSUE TITLE
Bot skips instructions causing the bot to fail

### ISSUE THREAD

**Issue by jason00-r** [linus-jansson/btd6farmer#23]
the bot lost at round 78 and is not handling the defeat screen. options for home, restart, review map, and continue. and the cmd window just prints the following while it clicks around missing any actual buttons:
```
selecting map
Game start
selecting map
Game start
selecting map
Game start
selecting map
Game start
```

map review:
![image](https://user-images.githubusercontent.com/16599377/180080641-f94a14d1-0ee8-4a50-a826-fe60f740f85e.png)

defeat screen: 
![image](https://user-images.githubusercontent.com/16599377/180080412-c1bc5758-d6f0-47ca-bcfc-744be3a5f38e.png)

**Comment by linus-jansson** [comment]
Should be fixed now, realized that I never added a button for the defeat screen. So it just handled it as a victory and thought it went to back to the main menu..

Were there any reason that you got defeated?

**Comment by jason00-r** [comment]
> Should be fixed now, realized that I never added a button for the defeat screen. So it just handled it as a victory and thought it went to back to the main menu..
> 
> Were there any reason that you got defeated?

I'm not sure why I lost that game. I didn't look too closely at the towers before I went home. If it happens again I will note tower upgrades to see if something was not done correctly.

**Comment by jason00-r** [comment]
> Should be fixed now, realized that I never added a button for the defeat screen. So it just handled it as a victory and thought it went to back to the main menu..
> 
> Were there any reason that you got defeated?

just lost on round 52 this time... here is tower levels:

dart monkey 0-0-2 
dart monkey 0-0-0
monkey village 0-0-1 and 0-0-2
super monkey 0-0-0
obyn level 10, almost 11
alchemist 3-0-0
both monkey subs 2-0-3

screenshot of placed towers and the bloons that made it by shown on the defeat screen:

![image](https://user-images.githubusercontent.com/16599377/180356102-8155856d-b575-4c8b-9012-793d157e82b0.png)
![image](https://user-images.githubusercontent.com/16599377/180356116-91057854-0c33-42dd-b702-0e844cd3745c.png)

I didn't use the newest commit that fixed clicking off the defeat screen since I wanted to be able to look at the towers if I lost again.

**Comment by linus-jansson** [comment]
It seems like its skippping instructions, mentioned in #21 

The problem I have with debugging this is that I cannot reproduce it.. 
Could still be a possibility that its tesseract that is having issues with retrieving the current round.. 

Have you played multiple matches before this happens or is this after the first try? If its your first match that you are loosing on, could you please check if rounds 14, 21 and 22 anything happens & also 45, 46, 47 and 49 which should upgrade the super monkey.. 

In #21 aranthel09 mentions that the bot opens the menu but doesn't upgrade.. Could you check if you get the same behaviour?

**Comment by jason00-r** [comment]
> It seems like its skippping instructions, mentioned in #21
> 
> The problem I have with debugging this is that I cannot reproduce it.. Could still be a possibility that its tesseract that is having issues with retrieving the current round..
> 
> Have you played multiple matches before this happens or is this after the first try? If its your first match that you are loosing on, could you please check if rounds 14, 21 and 22 anything happens & also 45, 46, 47 and 49 which should upgrade the super monkey..
> 
> In #21 aranthel09 mentions that the bot opens the menu but doesn't upgrade.. Could you check if you get the same behaviour?
> 
> EDIT: If you are unsure about when the bot is loosing, you can run the bot with the `--debug` flag by either putting it in the `Run.bat` file or as an argument when running from the CMD. It will generate a statsfile that tells you how many matches the bot has lost and won.
> 
> Also if you could try if this problem persists in the latest release

For me this behavior happens after many many rounds, like if I leave the bot unattended for 4+ hours, and even then it seems random when/if it will happen. 

I will try the latest release and run with the debug flag to see if this still happens.

**Comment by linus-jansson** [comment]
> For me this behavior happens after many many rounds, like if I leave the bot unattended for 4+ hours, and even then it seems random when/if it will happen.
> 
> I will try the latest release and run with the debug flag to see if this still happens.

Will try to get to the bottom of this with #25

**Comment by linus-jansson** [comment]
I will close this because the main issue is fixed and the other mentioned issue is in #21
